src/spec_utils.jl

```diff
@@ -1,3 +1,25 @@
+struct ObjectProxy
+    params::Dict
+end
+
 abstract type AbstractVegaSpec end
 
 Base.copy(spec::T) where {T <: AbstractVegaSpec} = T(copy(spec.params))
+
+const ObjectLike = Union{AbstractVegaSpec, ObjectProxy}
+
+getparams(spec) = getfield(spec, :params)
+
+Base.propertynames(spec::ObjectLike) =
+    [Symbol(key) for key in keys(spec.params)]
+
+function Base.getproperty(spec::ObjectLike, name::Symbol)
+    name === :params && return getparams(spec)
```

Comment:
Should we deprecate `.params`?

Reply:
Yes, I think so! That really should be an internal implementation detail, I think.